lib/javascript/JavascriptModulesPlugin.js

```diff
@@ -578,10 +578,20 @@ class JavascriptModulesPlugin {
 					const innerStrict = !allStrict && m.buildInfo.strict;
 					const iife = innerStrict || inlinedModules.size > 1 || chunkModules;
 					if (iife) {
-						source.add("!function() {\n");
-						if (innerStrict) source.add('"use strict";\n');
-						source.add(renderedModule);
-						source.add("\n}();\n");
+						if (
+							!runtimeRequirements.has(RuntimeGlobals.thisAsExports) &&
```

Comment:
```suggestion
```

Actually I told you something wrong. That doesn't make sense here in any way. It's an IIFE and doesn't pass any `this` value anyway.